### ISSUE TITLE
Ingestion tasks stuck on PENDING for a while

### ISSUE THREAD

**Issue by shachar-ash** [apache/druid#11196]
### Affected Version

We are using version 0.21.0 

### Description

When running ingestion tasks we see that some of our tasks are stuck at "PENDING" state before starting even though there are no running tasks for the same interval or even the same datasource. Also, there are plenty of available slots in the middle-managers where we saw this error.

- Cluster size
   We have ~30 middle-managers of different types and JVMs - We have different middle-managers for native ingestion/EMR ingestion/Kafka ingestion.
   We run ~20k ingestion tasks a day.

- Configurations in use
   We use a custom JavaScript affinity to determine where to navigate each task.

- The error message or stack traces encountered. Providing more context, such as nearby log messages or even entire logs, can be helpful.
  We didn't encounter any error, the task is stuck at PENDING state for a while, no task log is available and we couldn't find any indicative message in the middle-manager logs.

- Any debugging that you have already done
* Searched middle-manager logs
* Searched task logs

Any help would be really appreciated!

Thanks,
Shachar

**Comment by jihoonson** [comment]
Hi @shachar-ash, thank you for the report. The pending task status means that the Overlord thinks there is no task slot available for the given task for some reason. Can you look at the overlord logs and see if there is any interesting logs? Also, do you see any pattern in those pending tasks? Are they all different task types (`index_kafka`, `index_parallel`, `index_hadoop`)?

**Comment by shachar-ash** [comment]
Hi @jihoonson, first of all, thanks for the quick response, I really appreciate it!

I'm seeing a lot of these logs in overlord logs:

> 2021-05-04T16:59:03,632 INFO [TaskQueue-Manager] org.apache.druid.indexing.overlord.RemoteTaskRunner - Assigned a task[my_task_123_20210504_2021-04-04_2021-05-04T16:59:01_aeab9d4c-40d6-4479-a717-cbdd199e9e43] that is already pending!

We currently don't use index_parallel, earlier this week we upgraded our cluster and it wasn't available for us before.
It doesn't happen for `index_kafka` tasks, only for `index` and `index_hadoop` tasks.

We suffer from it the most in a middle-manager with only 1 worker, and therefore no tasks are running when it happens on it. 

Couldn't find a strong pattern so far, but I'll do my best to note every time it happened and update here when I'll have more information.
